src/rmkit/fb/fb.cpy

```diff
@@ -149,6 +151,12 @@ namespace framebuffer:
     virtual void set_screen_depth(int d):
       return
 
+    inline void _set_pixel(remarkable_color *dst, int x, int y, remarkable_color c):
+      *dst = self.dither(x, y, c);
```

Comment:
One of my initial attempts at this used a `Pattern` base class that could be used for dithering and patterns. It might be worth heading in that direction eventually -- this could be pretty similar to the "Brush" concept that a lot of UI toolkits use, e.g.

```cpp
class Brush {
public:
  virtual void set_pixel(Color *dst, int x, int y, Color c) = 0;
};


class Hatching : public Brush {
public:
  void set_pixel(Color *dst, int x, int y, Color c)
  {
    // similar to the ERASER brushes in do_dithering
    if ((x + y) % 3 == 0)
      *dst = c;
    else
      *dst = WHITE;
  }
}

class Stipple : public Brush {
public:
  float intensity = 1.0;
  void set_pixel(Color *dst, int x, int y, Color c)
  {
    // for charcoal-like brushes, similar to the dithering arg in do_dithering
    if (fast_rand() / FAST_RAND_MAX < intensity)
      *dst = c;
    // else don't' color at all
  }
};



```

Reply:
sounds reasonable. maybe there will be be worse perf by using vtable instead of inline but want to profile

Reply:
Yeah, the vtable should add 1 extra deref instruction (at least it does on x86, I don't know how to read ARM assembly well enough). But it should make it possible to make the current `do_dithering` into a brush (or even several different brushes). I think it might even save instructions in the most common case of drawing with a normal solid fill since you wouldn't have to go through the do_dithering switch statement.